extensions/mssql/src/objectExplorerNodeProvider/objectExplorerNodeProvider.ts

```diff
@@ -0,0 +1,348 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as sqlops from 'sqlops';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+import { ProviderBase } from './providerBase';
+import { Connection } from './connection';
+import * as utils from '../utils';
+import { TreeNode } from './treeNodes';
+import { ConnectionNode, TreeDataContext, ITreeChangeHandler } from './hdfsProvider';
+import { IFileSource } from './fileSources';
+import { AppContext } from '../appContext';
+import * as constants from '../constants';
+
+const outputChannel = vscode.window.createOutputChannel(constants.providerId);
+interface IEndpoint {
+	serviceName: string;
+	ipAddress: string;
+	port: number;
+}
+
+export class MssqlObjectExplorerNodeProvider extends ProviderBase implements sqlops.ObjectExplorerNodeProvider, ITreeChangeHandler {
+	public readonly supportedProviderId: string = constants.providerId;
+	private sessionMap: Map<string, Session>;
+	private expandCompleteEmitter = new vscode.EventEmitter<sqlops.ObjectExplorerExpandInfo>();
+
+	constructor(private appContext: AppContext) {
+		super();
+
+		this.sessionMap = new Map();
+		this.appContext.registerService<MssqlObjectExplorerNodeProvider>(constants.ObjectExplorerService, this);
+	}
+
+	handleSessionOpen(session: sqlops.ObjectExplorerSession): Thenable<boolean> {
+		return new Promise((resolve, reject) => {
+			if (!session) {
+				reject('handleSessionOpen requires a session object to be passed');
```

Comment:
Do we need to localize?

Reply:
All existing reject messages don't using localize. Do we want to change all of them